### ISSUE TITLE
CICD - Build a pipeline to automatically deploy changes

### ISSUE THREAD

**Issue by AndyBoWu** [treasurenetprotocol/docs#7]
**CONTEXT**
We need a highly automated pipeline to automatically deploy the changes to the doc server once a PR to the main branch is approved and merged. We have a teamcity based pipeline but it is associated with a feature branch, also we don't want to pay more efforts to manage a teamcity server cluster.

DONE WHEN
1/ pipeline is implemented using GitHub Action
2/ the pipeline is associated with the main branch
3/ documentation delivered
4/ + demo

NOTES
1/ you might need server info
2/ we may need another ticket to manage this server using IaC
3/ Andy will create GHA template repos

**Comment by AndyBoWu** [comment]
I am thinking to use GH Pages to host the entire project. Let's chat when available, @jhd2best

**Comment by AndyBoWu** [comment]
This is done.@jhd2best

sample job: https://github.com/treasurenetprotocol/treasurenetprotocol.github.io/actions/runs/6899832314
